test/functional/test_framework/util.py

```diff
@@ -480,6 +480,28 @@ def create_confirmed_utxos(fee, node, count):
     return utxos
 
 
+def chain_transaction(node, parent_txids, vouts, value, fee, num_outputs):
```

Comment:
Kind of a nit, but what are you thoughts on taking `utxos` as an input param instead of both `parent_txids` and `vouts`? Then you can get rid of:
```python
for (txid, vout) in zip(parent_txids, vouts):
         inputs.append({'txid' : txid, 'vout' : vout})
```
as `utxos` would be of the form: `[{ 'txid': '9d5b026e3dfe0d6c9ac95fc21d7f5802e046d85066c6df92fdb407471ace3b0c', 'vout': 0 }, ...]`

To me it seems like this slightly simplifies this function, and is a bit more understandable. And it also probably simplifies the calling code a bit as it seems a lot of times your parsing `parent_txids` and `vouts` from utxos before you call

I know it diverges a bit from existing code, but maybe it could be a refactor commit if this is a good time to do it. More of a nit to me

Reply:
Thanks for the suggestion, I think taking `utxos` as an input would be a good idea, leading to a cleaner interface and also improving readability. Right now though, at many places it would also need to introduce longer code in test, crafting together the utxo dictionary objects.

Probably this is a good candidate for a follow-up PR that also takes care of the return value? I think the same utxo structure you suggested could then also be used as returned type. I.e. an utxo object created by `chain_transaction` could be directly fed as new input utxo in places where it is called in loops. Another idea would be to drop the amount parameter and simply always spend the sum of all utxo amounts (didn't check though yet if there is a place where less is spent).

Reply:
Ok, agreed that this should be a follow up PR if anything. I'll circle back around to this some time after this PR is merged, and if no one has looked at it then I may take a whack at it